grpc/src/main/java/com/replica/RouterImpl.java

```diff
@@ -216,171 +76,27 @@ public void info(InfoRequest request, StreamObserver<InfoReply> responseObserver
-                Trip.WalkLeg thisLeg = (Trip.WalkLeg) leg;
-                String travelSegmentType;
-                String legMode = "foot";
-
-                // Assign proper ACCESS/EGRESS/TRANSFER segment type based on position of walk leg in list
-                if (i == 0) {
-                    travelSegmentType = "ACCESS";
-                    accessExists = true;
-                    legMode = ghPtRequest.getAccessProfile();
-                } else if (i == legs.size() - 1) {
-                    travelSegmentType = "EGRESS";
-                    egressExists = true;
-                    legMode = ghPtRequest.getEgressProfile();
-                } else {
-                    // Note: transfer legs are always walking (mode "foot")
-                    travelSegmentType = "TRANSFER";
-                }
-                path.getLegs().add(RouterConverters.toCustomStreetLeg(thisLeg, travelSegmentType, legMode));
-            } else if (leg instanceof Trip.PtLeg) {
-                Trip.PtLeg thisLeg = (Trip.PtLeg) leg;
-
-                long startTime = System.currentTimeMillis();
-                CustomPtLeg customPtLeg = RouterConverters.toCustomPtLeg(thisLeg, gtfsFeedIdMapping, gtfsLinkMappings, gtfsRouteInfo);
-                double durationSeconds = (System.currentTimeMillis() - startTime) / 1000.0;
-                String[] tags = MetricUtils.applyCustomTags(new String[0], customTags);
-                MetricUtils.sendInternalRoutingStats(statsDClient, tags, durationSeconds, "to_custom_pt_leg_seconds");
-
-                path.getLegs().add(customPtLeg);
-            }
-        }
+    @Override
+    public void routeCustom(CustomRouteRequest request, StreamObserver<StreetRouteReply> responseObserver) {
+        customStreetRouter.routeCustom(request, responseObserver);
+    }
 
-        // TODO: can we remove this clause? Or does the duplicate ID issue still exist for access/egress legs?
-        if (accessExists && egressExists) {
-            // ACCESS legs contains stable IDs for both ACCESS and EGRESS legs for some reason,
-            // so we remove the EGRESS leg IDs from the ACCESS leg before storing the path
-            CustomStreetLeg accessLeg = (CustomStreetLeg) path.getLegs().get(0);
-            CustomStreetLeg egressLeg = (CustomStreetLeg) path.getLegs().get(path.getLegs().size() - 1);
-            accessLeg.stableEdgeIds.removeAll(egressLeg.stableEdgeIds);
-        }
+    @Override
+    public void routeIsochrone(IsochroneRouteRequest request, StreamObserver<IsochroneRouteReply> responseObserver) {
+        isochroneRouter.routeIsochrone(request, responseObserver);
+    }
 
-        // Calculate correct distance incorporating foot + pt legs
-        path.setDistance(path.getLegs().stream().mapToDouble(l -> l.distance).sum());
+    @Override
+    public void routePt(PtRouteRequest request, StreamObserver<PtRouteReply> responseObserver) {
+        transitRouter.routePt(request, responseObserver);
```

Comment:
I think you only construct the transitRouter if ptRouter is passed in? Is it worth confirming transitRouter isn't null here, or is exception ok for handling unexpected request?

Reply:
Good callout! This is a case that probably warrants some real error logic even though the way we build routers (via model repo code, mostly) would make it tough to build a router intended for transit routing that doesn't include GTFS

Added some error logic for here + the PT isochrone endpoint, to avoid the misleading NPE that would've occurred without the null check